components/supervisor/frontend/src/ide/heart-beat.ts

```
@@ -16,14 +16,23 @@ export const track = (w: Window) => {
 }
 
 let toCancel: DisposableCollection | undefined;
-export function schedule(instanceId: string): void {
+export function schedule(workspaceId: string, instanceId: string, sessionId: string): void {
     if (toCancel) {
         return;
     }
     toCancel = new DisposableCollection()
     const sendHeartBeat = async (wasClosed?: true) => {
         try {
             await window.gitpod.service.server.sendHeartBeat({ instanceId, wasClosed });
+            wasClosed && await window.gitpod.service.server.trackEvent({
```

```ts
if (wasClosed) {
 window.gitpod.service.server.trackEvent({
                event: 'ide_close_signal',
                properties: {
                    workspaceId,
                    instanceId,
                    sessionId,
                    clientKind: 'supervisor-frontend'
                }
            });
}
```

we should not await analytics and swallow errors